### DISCUSSION TITLE
Disable paged output in `aws` command

### DISCUSSION THREAD

**Discussion by taylormoyle** [ohmyzsh/ohmyzsh#10683] [Help] [answered]
Is there a way to disable this object view? Currently I have to press Q over and over, and I would rather have them listed out like a normal bash window does. 

<img width="287" alt="image" src="https://user-images.githubusercontent.com/25063141/153499015-46ffaf83-acd4-4bd7-a903-e4c812ffaa67.png">

**Comment by mcornella** [comment, marked as answer]
What command is that? You're probably seeing `less`, which is a pager. Depending on the command you might be able to configure the output to not use a pager.

**Reply by taylormoyle** [reply]
I'm running through a series of aws commands, such as `aws s3api create-bucket`, `aws lambda create-function`, and `aws stepfunctions create-state-machine`. 

When I disable zsh by commenting out this line `# source $ZSH/oh-my-zsh.sh`, I get the standard sequential output without having to press Q.

**Reply by mcornella** [reply]
https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html

**Reply by taylormoyle** [reply]
That flag didn't work, and it doesn't paginate by default when I use standard bash, which led me to believe it was some setting in zsh or oh-my-zsh.

**Reply by mcornella** [reply]
The probable culprit is OMZ setting `$PAGER`. Try with `unset PAGER` (may have side effects).

**Reply by taylormoyle** [reply]
I tried, seemed too have no effect. When I echo $PAGER I don't get a response.

**Reply by mcornella** [reply]
The docs say the pager is used by default in awscli v2. If `$PAGER` is set, the output is paged. You can turns this off with `cli_pager=` in a profile or setting `AWS_PAGER=""`: https://docs.aws.amazon.com/cli/latest/userguide/cli-usage-pagination.html#cli-usage-pagination-clientside.

**Reply by taylormoyle** [reply]
That did it! I set `cli_pager=` in `~/.aws/config` and now it prints as I intended.

Thanks!
